src/tsickle.ts

```diff
@@ -2079,6 +2084,56 @@ function addClutzAliases(
   return dtsFileContent;
 }
 
+/**
+ * Returns 2 strings specifying the generic type arguments for the symbol.  The constrained params
+ * include any `T extends foo` arguments, the regular params are just a list of the type symbols,
+ * since we need the constraints on the LHS of the alias declaration, but can't have them on the
+ * RHS.
+ */
+function getGenericTypeParameters(symbol: ts.Symbol): {params: string, constrainedParams: string} {
+  if (!symbol.declarations) {
+    return {params: '', constrainedParams: ''};
+  }
+
+  if (!symbol.declarations[0]) {
+    return {params: '', constrainedParams: ''};
+  }
+
+  const declaration = symbol.declarations[0];
+
+  if ([
+        ts.SyntaxKind.FunctionDeclaration, ts.SyntaxKind.ConstructorKeyword,
+        ts.SyntaxKind.ClassDeclaration, ts.SyntaxKind.InterfaceDeclaration,
+        ts.SyntaxKind.TypeAliasDeclaration
+      ].indexOf(declaration.kind) === -1) {
+    return {params: '', constrainedParams: ''};
+  }
+
+  const declarationWithTypeParameters: ts.DeclarationWithTypeParameters =
+      declaration as ts.DeclarationWithTypeParameters;
+
+  if (!declarationWithTypeParameters.typeParameters) {
+    return {params: '', constrainedParams: ''};
+  }
+
+  const paramList: string[] = [];
+  const constrainedParamList: string[] = [];
+  for (const param of declarationWithTypeParameters.typeParameters) {
+    if (param.constraint) {
+      constrainedParamList.push(`${param.name.text} extends ${
+          (param.constraint as ts.TypeReferenceNode).typeName.getText()}`);
+    } else {
+      constrainedParamList.push(param.name.text);
+    }
+    paramList.push(param.name.getText());
+  }
+
+  const params = `<${paramList.join(',')}>`;
```

Comment:
nit, I think the < > and ',' belong in the emit code, and this helper should emit {param: string[], paramWithConstrain: string[]}

Reply:
I had it that way originally - I don't like how that requires additional variables, or a ternary expression at every usage.